docs/spec/database.md

```diff
@@ -3,17 +3,22 @@ topic: Database
 date: 2025-05-30
 discussion: https://github.com/orgs/k8tre/discussions/9
 k8tre-statements:
-  spec: K8TRE Specification-conformant apps shall allow the use of the default DB. A K8TRE should integrate with an organisation's existing databases where appropriate.
-  refimpl: The K8TRE Reference Implementation includes a default Postgres DB, for the general use of apps. CloudNativePG used with ArgoCD to configure and manage this  on-cluster DB.
+  spec: The TRE must offer a Kubernetes-native mechanism for deploying relational databases on the cluster, and optionally can also allow microservices to connect to off-cluster databases.
```

Comment:
Why?

This goes to the heart of what we're trying to achieve with the spec. You only need a database if you need a database.
- the reference implementation needs one for its core
- other apps might need one (but they shouldn't use the core one)

```suggestion
  spec: Databases should be treated as attached resources and may be deployed on the cluster, or apps may connect to off-cluster databases. Apps should not share databases.
```